ios/Classes/TiNfcNfcAdapterProxy.m

```diff
@@ -105,13 +186,61 @@ - (void)readerSession:(NFCNDEFReaderSession *)session didInvalidateWithError:(NS
         [_nNdefInvalidated call:@[ @{
           @"cancelled" : @(error.code == 200),
           @"message" : [error localizedDescription],
-          @"code" : NUMINTEGER([error code])
+          @"code" : NUMINTEGER([error code]),
+          @"type" : NDEF_READER_SESSION
+        } ]
+                     thisObject:self];
+      },
+      NO);
+}
+
+#pragma mark NFCReaderSessionDelegate
+
+- (void)tagReaderSession:(NFCTagReaderSession *)session didInvalidateWithError:(NSError *)error
+{
+  _nfcTagReadersession = nil;
+
+  if (!_nNdefInvalidated) {
+    return;
+  }
+  TiThreadPerformOnMainThread(
+      ^{
+        [_nNdefInvalidated call:@[ @{
+          @"cancelled" : @(error.code == 200),
+          @"message" : [error localizedDescription],
+          @"code" : NUMINTEGER([error code]),
+          @"type" : NFC_TAG_READER_SESSION
         } ]
                      thisObject:self];
       },
       NO);
 }
 
+- (void)tagReaderSessionDidBecomeActive:(NFCTagReaderSession *)session
+{
+
+  if (![self _hasListeners:@"tagReaderSessionDidBecomeActive"]) {
+    return;
+  }
+  [self fireEvent:@"tagReaderSessionDidBecomeActive"
+       withObject:@{
+         @"type" : NFC_TAG_READER_SESSION
+       }];
+  [self fireEvent:@"tagReaderSessionDidBecomeActive"];
+}
+
+- (void)tagReaderSession:(NFCTagReaderSession *)session didDetectTags:(NSArray<__kindof id<NFCTag>> *)tags
+{
+  if (![self _hasListeners:@"didDetectTags"]) {
+    return;
+  }
+  [self fireEvent:@"didDetectTags"
+       withObject:@{
+         @"tag" : [[TiNfcNativeTagTechnologyProxy alloc] _initWithPageContext:[self pageContext] andSession:_nfcTagReadersession andTag:tags],
```

Comment:
@anil-shukla-axway , Here in param ```tag``` you are sending TagTechonlogyProxy object but you should send array of ``TiNfcTagProxy``` object

Reply:
@vikas-goyal-axway  the event didDetectTag is return object of Native tags so shouldn't we be sending the data of native tag tech in our event.